src/sentry/integrations/github/integration.py

```diff
@@ -177,11 +177,24 @@ def get_repositories(
     def search_issues(self, query: str) -> Mapping[str, Sequence[Mapping[str, Any]]]:
         return self.get_client().search_issues(query)
 
+    def source_url_matches(self, url: str) -> bool:
```

Comment:
For all these functions,  can you add test cases?

Reply:
@AniketDas-Tekky Done! I've added some test cases.